Type error: Type 'Record<string, any>[]' is not assignable to type 'ExportData[]'.

Creating an optimized production build ...
Generated 9 documents in .contentlayer
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
 ✓ Compiled successfully
   Linting and checking validity of types ...
Failed to compile.

./src/components/analyses/prijsherziening/PrijsherzieningDashboard.tsx:336:15
Type error: Type 'Record<string, any>[]' is not assignable to type 'ExportData[]'.
  Type 'Record<string, any>' is missing the following properties from type 'ExportData': label, value

  334 |           <div className="flex items-center gap-2">
  335 |             <ExportButtons
> 336 |               data={chartData.map((row) => {
      |               ^
  337 |                 // Build data object with all selected components
  338 |                 const dataObj: Record<string, any> = {
  339 |                   label: row.label,
Error: Process completed with exit code 1.

--- a/embuild-analyses/src/components/analyses/prijsherziening/PrijsherzieningDashboard.tsx
+++ b/embuild-analyses/src/components/analyses/prijsherziening/PrijsherzieningDashboard.tsx
@@ -184,25 +184,6 @@ export function PrijsherzieningDashboard() {
     })
   }, [chartData, selectedComponentList])
 
-  // Prepare export data - flatten multi-series data for ExportButtons
-  const exportData = React.useMemo(() => {
-    const flattened: Array<{ label: string; value: number; periodCells: Array<string | number> }> = []
-
-    chartData.forEach(row => {
-      selectedComponentList.forEach(comp => {
-        const value = row[comp]
-        if (typeof value === "number") {
-          flattened.push({
-            label: `${row.label} - ${comp}`,
-            value: value,
-            periodCells: [row.year, row.month, comp],
-          })
-        }
-      })
-    })
-
-    return flattened
-  }, [chartData, selectedComponentList])
 
   // Calculate latest values
   const latestValues = React.useMemo(() => {
@@ -351,21 +332,26 @@ export function PrijsherzieningDashboard() {
           <div className="flex items-center gap-2">
             <ExportButtons
               data={chartData.map((row) => {
-                // For single component, use value field; for multiple, set to 0 (CSV export uses all columns)
-                const value = selectedComponentList.length === 1
-                  ? (row[selectedComponentList[0]] as number) || 0
-                  : 0
-
-                return {
+                // Base data structure
+                const baseData: {
+                  label: string
+                  value: number
+                  periodCells: Array<string | number>
+                  [key: string]: string | number | Array<string | number>
+                } = {
                   label: row.label,
-                  value: value,
+                  value: selectedComponentList.length === 1
+                    ? (row[selectedComponentList[0]] as number) || 0
+                    : 0,
                   periodCells: [row.label],
-                  // Add dynamic component columns
-                  ...selectedComponentList.reduce((acc, comp) => {
-                    acc[comp] = (row[comp] as number) || 0
-                    return acc
-                  }, {} as Record<string, number>)
                 }
+
+                // Add each component as a separate column for multi-series CSV export
+                selectedComponentList.forEach(comp => {
+                  baseData[comp] = (row[comp] as number) || 0
+                })
+
+                return baseData
               })}
               periodHeaders={["Periode"]}
               title="Prijsherzieningsindex I-2021"

--- a/embuild-analyses/src/components/analyses/shared/ExportButtons.tsx
+++ b/embuild-analyses/src/components/analyses/shared/ExportButtons.tsx
@@ -14,6 +14,7 @@ type ExportData = {
   label: string
   value: number
   periodCells?: Array<string | number>
+  [key: string]: string | number | Array<string | number> | undefined
 }
 
 interface ExportButtonsProps {
@@ -75,13 +76,36 @@ export function ExportButtons({
     metadata.push(`# Analyse: ${baseUrl}/analyses/${slug}/`)
     metadata.push(`#`)
 
+    // Detect additional columns (beyond standard label, value, periodCells)
+    const standardKeys = new Set(["label", "value", "periodCells"])
+    const additionalColumns: string[] = []
+
+    if (data.length > 0) {
+      const firstRow = data[0]
+      for (const key in firstRow) {
+        if (!standardKeys.has(key) && typeof firstRow[key] === "number") {
+          additionalColumns.push(key)
+        }
+      }
+    }
+
     // Build CSV headers
-    const headers = [...periodHeaders, valueLabel]
+    const headers = additionalColumns.length > 0
+      ? [...periodHeaders, ...additionalColumns]
+      : [...periodHeaders, valueLabel]
 
     // Build CSV rows
     const rows = data.map((row) => {
       const periodValues = row.periodCells ?? [row.label]
-      return [...periodValues, row.value].join(",")
+
+      if (additionalColumns.length > 0) {
+        // Multi-column export: include all additional columns
+        const columnValues = additionalColumns.map(col => row[col] ?? "")
+        return [...periodValues, ...columnValues].join(",")
+      } else {
+        // Single value export: use the value field
+        return [...periodValues, row.value].join(",")
+      }
     })
 
     const csv = [...metadata, headers.join(","), ...rows].join("\n")

